### PR TITLE
Pass platformId to isBot hook to allow loading images on universal

### PR DESCRIPTION
**Pull request by intellix**
Partially addresses https://github.com/tjoskar/ng-lazyload-image/issues/414

### CODE REVIEW

File: src/lazyload-image.directive.ts [hunk 1]
```diff
@@ -25,14 +25,14 @@ export class LazyLoadImageDirective implements OnChanges, AfterContentInit, OnDe
   private ngZone: NgZone;
   private scrollSubscription?: Subscription;
   private hooks: HookSet<any>;
-  private platformId: Object;
+  private platformId: object;
 
-  constructor(el: ElementRef, ngZone: NgZone, @Inject(PLATFORM_ID) platformId: Object, @Optional() @Inject('options') options?: ModuleOptions) {
+  constructor(el: ElementRef, ngZone: NgZone, @Inject(PLATFORM_ID) platformId: object, @Optional() @Inject('options') options?: ModuleOptions) {
```
Comment (tjoskar) [inline src/lazyload-image.directive.ts]: It should be `Object` here. `@Inject(PLATFORM_ID)` will resolve a `Object` and `isPlatformServer` accepts an `Object`. This row also makes the build to fail right now with the following error: `Metadata collected contains an error that will be reported at runtime: Expression form not supported` but if I change it to `Object` it works fine.

File: src/hooks-factory.ts [hunk 2]
```diff
@@ -3,11 +3,11 @@ import { ssrPreset } from './ssr-preset';
 import { HookSet, ModuleOptions } from './types';
 import { getNavigator } from './util';
 
-export function cretateHooks<E>(options?: ModuleOptions<E>): HookSet<any> {
+export function createHooks<E>(options?: ModuleOptions<E>, platformId?: object): HookSet<any> {
```
Comment (tjoskar) [inline src/hooks-factory.ts]: Nice typo fix.

File: src/types.ts [hunk 3]
```diff
@@ -51,7 +51,7 @@ export type SetErrorImageFn = (args: SetErrorImageProps) => void;
 export type SetupFn = (attributes: Attributes) => void;
 export type FinallyFn = (attributes: Attributes) => void;
 export type GetObservableFn<E> = (attributes: Attributes) => Observable<E>;
-export type IsBotFn = (navigator?: Navigator) => boolean;
+export type IsBotFn = (navigator?: Navigator, platformId?: object) => boolean;
```
Comment (tjoskar) [inline src/types.ts]: I think `platformId` should be non-optional:
```suggestion
export type IsBotFn = (navigator: Navigator | undefined, platformId: Object) => boolean;
```

Comment (tjoskar) [reply src/types.ts]: And update `createHooks` to something like:
```ts
export function createHooks<E>(platformId: Object, options?: ModuleOptions<E>): HookSet<any> {
```

File: src/lazyload-image.directive.ts (hunk 1 above)
Comment (intellix) [reply src/lazyload-image.directive.ts]: ahah, had this in mind: https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html

Didn't realise it was needed here. Will fix throughout

**Comment by tjoskar** [comment]
Thanks!

**Comment by tjoskar** [comment]
This is released in `ng-lazyload-image@7.0.1-0` 🎉

**Pull request merged**